src/NCalc.Async/AsyncExpression.cs

```diff
@@ -242,55 +242,40 @@ public bool HasErrors()
     {
         var parameterEnumerators = new Dictionary<string, IEnumerator>();
         int? size = null;
-        var results = new List<object?>();
 
-        foreach (var parameter in Parameters.Values)
+        foreach (var parameter in Parameters)
         {
-            if (parameter is not IEnumerable enumerable)
-                continue;
+            if (parameter.Value is IEnumerable enumerable)
+            {
+                var list = enumerable as List<object> ?? enumerable.Cast<object>().ToList();
+                parameterEnumerators.Add(parameter.Key, enumerable.GetEnumerator());
```

Comment:
You almost made it:

Wrong:
![image](https://github.com/ncalc/ncalc/assets/52143624/ff5cee7b-2610-402b-ae07-6c31e1d4c597)
![image](https://github.com/ncalc/ncalc/assets/52143624/cb1909ad-f298-43d2-991f-0638cf0ad424)